main.go

```diff
@@ -27,7 +53,41 @@ func main() {
 	// Env vars for the pizza oven server
 	serverPort := os.Getenv("SERVER_PORT")
 
+	// User specify which git provider to use
+	gitProvider := os.Getenv("GIT_PROVIDER")
+
+	// Initialize the database handler
 	pizzaOven := database.NewPizzaOvenDbHandler(databaseHost, databasePort, databaseUser, databasePwd, databaseDbName)
-	pizzaOvenServer := server.NewPizzaOvenServer(pizzaOven)
+
+	var pizzaGitProvider providers.GitRepoProvider
+	switch gitProvider {
+	case "cache":
+		sugarLogger.Infof("Initiating cache git provider")
+
+		// Env vars for the git provider
+		cacheDir := os.Getenv("CACHE_DIR")
+		minFreeDisk := os.Getenv("MIN_FREE_DISK_GB")
+
+		// Validates the provided minimum free disk int is parsable as a uint64
+		//
+		// TODO - should dynamically check file system bit size after compilation.
+		// 64 bit wide words should be fine for almost all use cases for now.
+		minFreeDiskUint64, err := strconv.ParseUint(minFreeDisk, 10, 64)
+		if err != nil {
+			sugarLogger.Fatalf(": %s", err.Error())
+		}
+
+		pizzaGitProvider, err = providers.NewLRUCacheGitRepoProvider(cacheDir, minFreeDiskUint64, sugarLogger)
+		if err != nil {
+			sugarLogger.Fatalf("Could not create a cache git provider: %s", err.Error())
+		}
+	case "memory":
```

Comment:
Should there be some sort of warning for using the memory option on large repos?

Reply:
Yeah, a quick warning log isn't a bad idea!